### PR TITLE
feat: add support for different revocation types for v2 and v3

### PR DESCRIPTION
**Pull request by cavacado**
this pr does:
- adds more values to the RevocationType enum for v2 and v3
- adds the optional ~~address~~ `location` field in the revocation object
- ~~adds tests for the introduced field.~~

### CODE REVIEW

File: src/2.0/schema/schema.json [hunk 1]
```diff
@@ -142,7 +147,12 @@
               "properties": {
                 "type": {
                   "type": "string",
-                  "enum": ["NONE"]
+                  "enum": ["NONE", "REVOCATION_STORE"]
+                },
+                "address": {
+                  "type": "string",
+                  "pattern": "^0x[a-fA-F0-9]{40}$",
+                  "description": "Smart contract address for Revocation Store type revocation"
```
Comment (cavacado) [inline src/2.0/schema/schema.json:155]: help me understand this revocationStore property here, what is the difference between this and the revocation object itself?
@yehjxraymond

Comment (Nebulis) [reply src/2.0/schema/schema.json:155]: Do you mean between the `revocation` key  and the "REVOCATION_STORE" value ?

File: src/2.0/schema/schema.json [hunk 2]
```diff
@@ -142,7 +147,12 @@
               "properties": {
                 "type": {
                   "type": "string",
-                  "enum": ["NONE"]
+                  "enum": ["NONE", "REVOCATION_STORE"]
+                },
+                "address": {
```
Comment (yehjxraymond) [inline src/2.0/schema/schema.json:152]: Might want to generalize this, we might have other methods like certificate revocation list etc in the future.

File: src/2.0/schema/schema.json (hunk 1 above)
Comment (cavacado) [reply src/2.0/schema/schema.json:155]: not really, this is the generated type for v2 schema for the type `Issuer`:
```ts
export interface Issuer {
    /**
     * Issuer's id, DID can be used
     */
    id?:            string;
    identityProof?: IdentityProof;
    /**
     * Issuer's name
     */
    name:        string;
    revocation?: Revocation;
    /**
     * Smart contract address of token registry
     */
    tokenRegistry?: string;
    /**
     * Smart contract address of document store
     */
    documentStore?: string;
    /**
     * Smart contract address of certificate store. Same as documentStore
     */
    certificateStore?: string;
    revocationStore?:  RevocationStore;
}
```
note that theres already a `revocation` key and it is exactly the same (value-wise) as the `revocationStore` key.

so was wondering why is there a need for the `revocationStore` key in the first place.

File: src/2.0/schema/schema.json (hunk 2 above)
Comment (cavacado) [reply src/2.0/schema/schema.json:152]: what do they look like? could you give examples so I can see how I can generalise this?

Comment (cavacado) [reply src/2.0/schema/schema.json:152]: will change to a more generic key to `location` as discussed with raymond

**Review by rjchow — APPROVED** [review]

File: src/2.0/schema/schema.json (hunk 1 above)
Comment (yehjxraymond) [reply src/2.0/schema/schema.json]: image

Think that the original consideration was that the revocation block will tell us what to use and then in the case of `REVOCATION_STORE` we find the revocationStore block. Additionally, seemed appropriate to have it at the same level at the `documentStore` and `certificateStore`

Seem like if we generalise the address->location, it should be able to support any type of revocation within that block. 

I'll suggest we remove `revocationStore`... v2 is quite fucked anyway lol

Comment (cavacado) [reply src/2.0/schema/schema.json]: got it

**Review by yehjxraymond — APPROVED** [review]
lgtm

**Pull request merged**